src/extension.ts

```diff
@@ -3,12 +3,30 @@ import { SetupGranitePage } from "./panels/setupGranitePage";
 import { Telemetry } from "./telemetry";
 
 export async function activate(context: ExtensionContext) {
+  // Initialize telemetry
   await Telemetry.initialize(context);
+
+  // Register the setup command
   const setupGraniteCmd = commands.registerCommand("vscode-granite.setup", async () => {
     await Telemetry.send("granite.commands.setup");
     SetupGranitePage.render(context.extensionUri, context.extensionMode);
   });
   context.subscriptions.push(setupGraniteCmd);
-  // TODO check initial startup status
-  return commands.executeCommand('vscode-granite.setup');
-}
\ No newline at end of file
+
+  // Check if the setup has already run using globalState
+  const hasRunBefore = context.globalState.get('hasRunSetup', false);
+
+  // Check if we are in dev mode using the OLLAMA_MOCK environment variable
+  const isDevMode = process.env.OLLAMA_MOCK === 'true';
```

Comment:
no, this is not enough to determine you're in dev mode. https://github.com/redhat-developer/vscode-granite/blob/main/.vscode/launch.json#L19 is also in dev mode. Right now, dev mode is inferred from Context.extensionMode, in https://github.com/redhat-developer/vscode-granite/blob/main/src/panels/setupGranitePage.ts#L71, but that's not satisfying either. You could run another VS Code extension in Dev Mode and that would make this one enter in dev mode as well.

What is needed is a refactoring, where isDevMode is moved to https://github.com/redhat-developer/vscode-granite/blob/main/src/commons/constants.ts
- `export const isDevMode = process.env.GRANITE_EXT_DEV_MODE === 'true';`
- add GRANITE_EXT_DEV_MODE env var to both launch configs in https://github.com/redhat-developer/vscode-granite/blob/main/.vscode/launch.json
- replace extensionMode by isDevMode in the constructor in https://github.com/redhat-developer/vscode-granite/blob/main/src/panels/setupGranitePage.ts#L53